### PR TITLE
Update CI/CD

### PR DESCRIPTION
**Pull request by zmraul**
Splits integration tests into different jobs and add them as requirements before release.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -38,4 +38,56 @@ jobs:
         with:
           provider: lxd
       - name: Run integration tests
-        run: tox -e integration
+        run: tox -e integration-charm
+
+  integration-provider:
+    name: Integration tests for provider (lxd)
+    runs-on: ubuntu-latest
```
Comment (marcoppenheimer) [inline .github/workflows/ci.yaml:45]: As we're editing this, could we add a `needs` block to all these int-tests based on Jon's suggestion in MM?

```yaml
  integration-provider:
    name: Integration tests for provider (lxd)
    runs-on: ubuntu-latest
    needs:
      - lint
      - unit-test
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Setup operator environment
        uses: charmed-kubernetes/actions-operator@main
        with:
          provider: lxd
      - name: Run integration tests
        run: tox -e integration-provider
```

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -19,11 +19,104 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: python -m pip install tox
+      - name: Run tests
+        run: tox -e unit
+
+  integration-charm:
+    name: Integration tests for (lxd)
+    runs-on: ubuntu-latest
```
Comment (marcoppenheimer) [inline .github/workflows/release.yaml:46]: As we're editing this, could we add a `needs` block to all these int-tests based on Jon's suggestion in MM?

```yaml
  integration-charm:
    name: Integration tests for charm (lxd)
    runs-on: ubuntu-latest
    needs:
      - lint
      - unit-test
      - lib-check
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Setup operator environment
        uses: charmed-kubernetes/actions-operator@main
        with:
          provider: lxd
      - name: Run integration tests
        run: tox -e integration-charm
```

**Review by marcoppenheimer — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (zmraul) [reply .github/workflows/ci.yaml:45]: I didn't catch the message. Is this to avoid running a bunch of time consuming tests when lint/unit doesn't pass?

**Review by paulomach — APPROVED** [review]
impressive

**Comment by zmraul** [comment]
CI is stuck

**Pull request closed without merging**